build.sc

```diff
@@ -1559,191 +1650,301 @@ object dev extends MillModule {
+
+    val renderedExamples: Seq[(os.SubPath, PathRef)] =
+      T.traverse(example.exampleModules)(m => T.task {
+        (m.millSourcePath.subRelativeTo(example.millSourcePath), m.rendered())
+      })()
+
+    for ((name, pref) <- renderedExamples) os.copy(
+      pref.path,
+      pagesWd / "example" / os.SubPath(s"$name.adoc"),
+      createFolders = true
+    )
+
+    val contribReadmes = T.traverse(contrib.contribModules)(m =>
+      T.task {
+        m.millModuleSegments.last.render -> m.readme()
+      }
+    )()
+
+    for ((name, pref) <- contribReadmes) os.copy(
+      pref.path,
+      pagesWd / "contrib" / s"${name}.adoc",
+      createFolders = true
+    )
+
+    PathRef(T.dest)
+  }
+  def supplementalFiles = T.source(millSourcePath / "supplemental-ui")
+  def devAntoraSources: Target[PathRef] = T {
+    val dest = T.dest
+    shared.mycopy(source().path, dest, mergeFolders = true)
+    val lines = os.read(dest / "antora.yml").linesIterator.map {
+      case l if l.startsWith("version:") =>
+        s"version: 'master'" + "\n" + s"display-version: '${millVersion()}'"
+      case l if l.startsWith("    mill-version:") =>
+        s"    mill-version: '${millVersion()}'"
+      case l if l.startsWith("    mill-last-tag:") =>
+        s"    mill-last-tag: '${millLastTag()}'"
+      case l => l
     }
-    def generatePages(authorMode: Boolean) = T.task {
-      // dependency to sources
-      source()
-      val docSite = T.dest
-      val playbook = docSite / "antora-playbook.yml"
-      val siteDir = docSite / "site"
-      os.write(
-        target = playbook,
-        data = githubPagesPlaybookText(authorMode)(),
-        createFolders = true
-      )
-      // check xrefs
-      runAntora(
-        npmDir = npmBase(),
-        workDir = docSite,
-        args = Seq(
-          "--generator",
-          "@antora/xref-validator",
-          playbook.last,
-          "--to-dir",
-          siteDir.toString(),
-          "--attribute",
-          "page-pagination"
-        ) ++
-          Seq("--fetch").filter(_ => !authorMode)
-      )
-      // generate site (we can skip the --fetch now)
-      runAntora(
-        npmDir = npmBase(),
-        workDir = docSite,
-        args = Seq(
-          playbook.last,
-          "--to-dir",
-          siteDir.toString(),
-          "--attribute",
-          "page-pagination"
-        )
+    os.write.over(dest / "antora.yml", lines.mkString("\n"))
+    PathRef(dest)
+  }
+  def githubPagesPlaybookText(authorMode: Boolean): Task[String] = T.task {
+    s"""site:
+       |  title: Mill
+       |  url: ${if (authorMode) s"${T.dest}/site/index.html" else Settings.docUrl}
+       |  start_page: mill::Intro_to_Mill.adoc
+       |
+       |content:
+       |  sources:
+       |    - url: ${if (authorMode) baseDir else Settings.projectUrl}
+       |      branches: []
+       |      tags: ["0.10.12", "0.9.12"]
+       |      start_path: docs/antora
```

Comment:
I'd expect some issue once we include tag `0.11.0-M8` which already moved `docs/antora` -> `docs`.

The workaround could be to switch from `tags` to `branches`, then create doc-branches from these current tags and in these branches, move the location to what we expect.

Reply:
Yes. We could also leave this content-source unchanged as `tags`, and add a new content type for the `branches` going forward.

Reply:
Yeah, I just re-discovered that we can use as many content sources for distinct tags/branch sets, so this should be not an issue. But I guess, we need to add all those generated/copied files and add them to the branches, so that antora can work it out. The alternative would be to switch to a folder-based approach, where we fetch all tags/branches ourselves and copy/generate the missing files.